src/core/ActivityInfoTree.cpp

```diff
@@ -272,4 +275,73 @@ void ActivityInfoTree::init()
     qmlRegisterSingletonType<QObject>("GCompris", 1, 0, "ActivityInfoTree", menuTreeProvider);
     qmlRegisterType<ActivityInfo>("GCompris", 1, 0, "ActivityInfo");
 }
+void ActivityInfoTree::beginsearch()
+{
+
+     m_menuTree = m_searched;
+     emit menuTreeChanged();
+     m_mainWindow = ApplicationInfo::getWindow()->findChild<QObject*>("menu");
+     QObject::connect(m_mainWindow,SIGNAL(textchanged(QString)),this,SLOT(display(QString)));
 
+}
+void ActivityInfoTree::display(QString text)
+{
+    int end,length,start = 0;
+
+
+
+       searchedtext = text;
+       end = m_menuTreeFulltemp.length();
+       length = text.length(); // length of the  string . i.e text in the textfield
+       m_menuTree.clear();
+
+       // templength = length of string that was there in the textfield before the present string
+       // if user erases a character from a string ,the search needs to be performed on all the activities
+       if(length < templength)
+       {
+
+           m_menuTreeFulltemp = m_menuTreeFull;
+           end = m_menuTreeFulltemp.length();
+       }
+
+       if(text.trimmed() == "")
```

Comment:
I was saying empty() instead of the operator==, the trimmed will still be needed (text.trimmed().empty() instead of text.trimmed() == "").
It avoids a string creation, the comparison and is clearer to understand.